client/src/components/base/ItemFilterBar/ItemFilterBar.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from 'react';
+import styled from '@emotion/styled';
+import { baemin, baeminFont, lightBlack } from '@/static/style/common';
+
+const filterName = ['추천순', '인기순', '최신순', '낮은가격순', '높은가격순'];
+
+type ItemFilterBarProps = {
+  handleFilter: (num: number) => void;
+  totalProductCount: number;
+};
+
+const ItemFilterBar = ({ handleFilter, totalProductCount }: ItemFilterBarProps) => {
+  const [index, setIndex] = useState(-1);
+
+  const onClickFilter = (e: React.MouseEvent) => {
+    const { target } = e;
+    if (!(target instanceof HTMLElement)) return;
+    setIndex(Number(target.dataset.idx));
+    handleFilter(Number(target.dataset.idx));
+  };
+
+  const createFilter = () => {
+    return filterName.map((itemName, idx) => (
+      <Fliter key={idx} onClick={onClickFilter} active={index === idx} data-idx={idx}>
+        {itemName}
+      </Fliter>
```

Comment:
```suggestion
  const onClickFilter = (index: number) => {
    setIndex(index);
    handleFilter(index);
  };

  const createFilter = () => {
    return filterName.map((itemName, idx) => (
      <Fliter key={`filter-item-${itemName}} onClick={onClickFilter} active={index === idx} index={idx}>
        {itemName}
      </Fliter>
      
      //Filter Component 분리
      
      interface FilterItemProps {
        index: number;
        onClickFilter: (index: number) => void;
      }
      
      const FilterItem = ({index, onClickFilter}: FilterItemProps) => {
        const onClick = () => {
          onClickFilter(index);
        }
      }
        
       return (
         <li onClick={onClick} >...</li>
       )
```
1. `dataset`을 사용하지 않고 `FilterItem`으로 분리해서 사용하는 방법을 소개드립니다. 
2. `key`값을 index로 사용하면 배열이 변경됐을 때(특히 unshift 상황) 리액트가 제대로 키값을 인지하지 못하는 문제가 있어서 최대한 유니크하게 만드는 것을 추천드립니다.

Reply:
오호 감사합니다 새로운 거 알아가네요 👍